### ISSUE TITLE
New Feature: Upload to Social Button and Download Video Button

### ISSUE THREAD

**Issue by JimTheMan** [JimTheMan/t2#15]
I was talking to someone about the app, and they gave me a really cool idea. We had already thought of putting in a "download video" button (either to download a single video clip or they entire sequence of clips as one video file), but that person said it would be cool to allow them to upload it right to youtube, snapchat, etc. Sort of like a "share" button that they have on youtube now.

**Comment by JimTheMan** [comment]
anyone think this would be cool? What do you want more: download or share socially?

**Comment by Faranch** [comment]
Sharing socially is great but its about making that initial contact one gets from seeing someone else post their vid look inviting. Maybe even a watermark on all the videos? Yet, we should have a share button where people can post on fb and other sites. Can endusers access an html video link if they wanted to post their vid on a forum or a site that the Share Button doesnt include?

Also, should we let people download other peoples videos or just their own? I think we shouldn't allow users to download videos for the MVP, unless its a large vid that pieces together everything. Maybe we can have a function that pieces a "endless story" together where everyone posts one bar

**Comment by taif95** [comment]
Yes, when i first thought of the cypher, one of the most important feature was making each video shareable to other social media outlets. I mentioned that we should have multiple channels and the "cypher" channel would allow users to record small videos (clips) corresponding to one another,  hence Cypher. After everyone records,  our app then merges all the clips into one whole video. Then the users should definitely have the option to post it where ever they like. But Only for the MVP,  when we actually launch we can try to work something out with other social media groups. 

As far as downloading, if possible we should have the app connect with the phone's photos/ gallery storage so it automatically saves after you publish a video. (Just like ig)

**Comment by shivpatel779** [comment]
@JimTheMan @Faranch 

yes it's good idea we will do with video share button in place of download button
